scripts/dev_scripts/integration_tests.sh

```diff
@@ -636,6 +638,32 @@ POLICY_EXPECTED=$WORKSPACE/tests/policy_engine/expected_results/policy_report.js
 $RUN_POLICY -f $POLICY_FILE -d "$WORKSPACE/output/macaron.db" || log_fail
 check_or_update_expected_output $COMPARE_POLICIES $POLICY_RESULT $POLICY_EXPECTED || log_fail
 
+# Testing the VSA generation feature
+# Running Macaron without config files
+echo -e "\n=================================================================================="
+echo "Run integration tests for VSA generation"
+echo -e "==================================================================================\n"
+TEST_CASE_DIR="$WORKSPACE/tests/vsa/integration/github_slsa-framework_slsa-verifier"
+OUTPUT_DIR="$TEST_CASE_DIR/output"
+
+rm -rf "$OUTPUT_DIR"  # Make sure we regenerate a fresh database every time.
+macaron --output "$OUTPUT_DIR" analyze \
+    --repo-path "https://github.com/slsa-framework/slsa-verifier" \
+    --digest 7e1e47d7d793930ab0082c15c2b971fdb53a3c95 \
+    --skip-deps || log_fail
+check_or_update_expected_output "$COMPARE_JSON_OUT" \
+    "$OUTPUT_DIR/reports/github_com/slsa-framework/slsa-verifier/slsa-verifier.json" \
+    "$TEST_CASE_DIR/slsa-verifier.json" || log_fail
+macaron --output "$OUTPUT_DIR" verify-policy \
+    --database "$OUTPUT_DIR/macaron.db" \
+    --file "$TEST_CASE_DIR/policy.dl" || log_fail
+check_or_update_expected_output "$COMPARE_POLICIES" \
+    "$OUTPUT_DIR/policy_report.json" \
+    "$TEST_CASE_DIR/policy_report.json" || log_fail
+python3 "$COMPARE_VSA" \
```

Comment:
Expected VSA payload files differ from the expected result files we have had so far since they only store the decoded payload of the VSA, not the whole VSA itself. Therefore, copying the result file to update the expected file would not work. I implemented the logic to update the expected payload file within the `compare_vsa.py` script. By default, the script does the comparison. If it runs with the `--update` flag, it switches to "update" mode and instead updates the expected payload file.

Reply:
Adjusted in e621ec3a2711205766450727cdd4736d2d70dcb3.